src/core/dfn.js

```diff
@@ -237,4 +237,37 @@ function addContractDefaults() {
   for (const dfn of exportableDfns) {
     dfn.dataset.export = "";
   }
+
+  // - Sets data-defines on well-known definition content patterns
+
+  // A dt with class dfn-desc (on in a dl with such a class)
+  // containing a definition
+  // indicates that the following dd or div element contains its prose content
+  /** @type NodeListOf<HTMLElement> */
+  const describedDTs = document.querySelectorAll(
+    "dl.dfn-desc dt:has(dfn[data-dfn-type]), dt.dfn-desc:has(dfn[data-dfn-type])"
+  );
+  for (const dt of describedDTs) {
+    const dfnId = dt.querySelector("dfn[id]").id;
```

Comment:
This code expects that there will always be one definition with an `id` attribute and yet you didn't use that in your filter to select the dt elements. If the `id` is always set, the `[id]` could be dropped. If not, the filter above should be updated to restrict to dfns that have an `id` attribute, I think.

Reply:
```suggestion
    const dfnId = dt.querySelector("dfn[data-dfn-type]").id;
```
good point (I had started by hooking on `[id]` but didn't think of updating this)